data_specification/enums/data_type.py

```diff
@@ -327,12 +327,50 @@ def encode_as_int(self, value):
         """ Returns the value as an integer, according to this type.
         """
         if self._apply_scale:
-            return int(round(decimal.Decimal(str(value)) * self.scale))
+            if not (self._min <= value <= self._max):
+                raise ValueError(
+                    "value {:f} cannot be converted to {:s}: out of range"
+                    .format(value, self.__doc__))
+            return int(round(decimal.Decimal(str(value)) * self._scale))
         if self._force_cast is not None:
             return self._force_cast(value)
         return value
 
+    def encode_as_numpy_int(self, value):
+        """ Returns the value as a numpy integer, according to this type.
+
+        .. note:
+            Only works with integer and fixed point data types.
+        """
+        return np.round(self.encode_as_int(value)).astype(self.struct_encoding)
+
+    def encode_as_numpy_int_array(self, array):
+        """ Returns the numpy array as an integer numpy array, according to \
+            this type.
+
+        :param array:
+        :type array: numpy.ndarray
+        :rtype array: numpy.ndarray
+        """
+        if self._apply_scale:
+            # pylint: disable=assignment-from-no-return
+            where = np.logical_or(array < self._min, self._max < array)
+            if any(where):
```

Comment:
Integration test failures are suggesting that this should be where.any() rather than any(where).

Reply:
Yes. I'd fouled up my interactive testing (when I checked through the logs, I saw that while I'd tried to check with multidimensional arrays, I passed the wrong thing into `any()` which made the case I was looking at work when it shouldn't have).